internal/controllers/topology/cluster/blueprint.go

```diff
@@ -92,5 +93,30 @@ func (r *Reconciler) getBlueprint(ctx context.Context, cluster *clusterv1.Cluste
 		blueprint.MachineDeployments[machineDeploymentClass.Class] = machineDeploymentBlueprint
 	}
 
+	// Loop over the machine pool classes in ClusterClass
+	// and fetch the related templates.
+	for _, machinePoolClass := range blueprint.ClusterClass.Spec.Workers.MachinePools {
+		machinePoolBlueprint := &scope.MachinePoolBlueprint{}
+
+		// Make sure to copy the metadata from the blueprint, which is later layered
+		// with the additional metadata defined in the Cluster's topology section
+		// for the MachinePool that is created or updated.
+		machinePoolClass.Template.Metadata.DeepCopyInto(&machinePoolBlueprint.Metadata)
+
+		// Get the infrastructure machine template.
+		machinePoolBlueprint.InfrastructureMachineTemplate, err = r.getReference(ctx, machinePoolClass.Template.Infrastructure.Ref)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get infrastructure machine template for %s, MachinePool class %q", tlog.KObj{Obj: blueprint.ClusterClass}, machinePoolClass.Class)
+		}
+
+		// Get the bootstrap machine template.
+		machinePoolBlueprint.BootstrapTemplate, err = r.getReference(ctx, machinePoolClass.Template.Bootstrap.Ref)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get bootstrap machine template for %s, MachinePool class %q", tlog.KObj{Obj: blueprint.ClusterClass}, machinePoolClass.Class)
```

Comment:
```suggestion
			return nil, errors.Wrapf(err, "failed to get bootstrap config template for %s, MachinePool class %q", tlog.KObj{Obj: blueprint.ClusterClass}, machinePoolClass.Class)
```
Please also change in the MD code in l.86